source/lambda/apiprocessor/redaction.py

```diff
@@ -0,0 +1,140 @@
+#  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES #
+#  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
+#  and limitations under the License.                                                                                #
+#####################################################################################################################
+
+import boto3
+import json
+import os
+from helper import S3Helper
+
+# the maximum number of exclusion lists we support, like legal, marketing, etc
+MAXIMUM_EXCLUSION_LISTS = 100
+
+# get the redacted items of this document
+def getDocumentRedaction(documentId, bucket):
+
+    redaction = {}
+    redaction['uuid'] = documentId
+    redaction['redactedItems'] = []
+    redaction['headers'] = []
+    redaction['footers'] = []
+    
+    # the redaction file may or may not exist
+    try:
+        data = S3Helper.readFromS3(bucket, 'public/' + documentId + '/redaction.json')
+        redaction = json.loads(data)
+    except Exception as e:
+        print("getDocumentRedaction exception: " + str(e))
+    except:
+        print("getDocumentRedaction exception")
+    
+    return redaction
+
+
+# retrieve the redacted items of this document
+def saveDocumentRedaction(documentId, bucket, body):
+
+    result = {}
+    result['status'] = False
+    
+    # save the document redaction, if it parses correctly
+    try:
+        redaction = json.loads(body)
+        
+        S3Helper.writeToS3(json.dumps(redaction),
+                           bucket,
+                           'public/' + documentId + '/redaction.json')
+    
+        result['status'] = True
+    
+    except Exception as e:
+        print("saveDocumentRedaction exception: " + str(e))
+    
+    return result
+
+
+# get the redacted items of this document
+def getRedactionGlobal(bucket):
+    
+    redactionglobal = {}
+    redactionglobal['labels'] = []
+    redactionglobal['exclusion'] = []
+    
+    #
+    # fetch redaction labels
+    #
+    try:
+        data = S3Helper.readFromS3(bucket, 'redactionglobal/labels.csv')
+        lines = data.splitlines()
+    
+        # remove csv header with column names
+        lines.pop(0)
+        
+        numOfLabels = len(lines)
+        
+        if numOfLabels > 0:
+            for i in range(0, numOfLabels):
+                tokens = lines[i].split(',')
+                label = {}
+                label['name'] = tokens[0]
+                label['display'] = tokens[1]
+                label['description'] = tokens[2]
+                redactionglobal['labels'].append(label)
+
+    # if the labels.csv file doesn't exists, then the labels are skipped
+    except Exception as e:
+        print("getRedactionGlobal exception while retrieving labels.csv: " + str(e))
+    except:
```

Comment:
What is this except statement for?
If the intention is to raise the error again, we can just use raise

Reply:
The labels.csv file in s3 is optional,  so trying to download it might fail with an exception if it doesn't exists, which is fine, hence the catch all.  We might consider adding a catch for that specific exception, so we can print a prettier log entry. 
  An alternative is to make a call first to see if it exists in s3,  then download it.  But that seems inefficient.

Reply:
My question is around why we have 2 except clauses here?
And do we expect the api flow to continue after the error or do we want to raise and end the flow?

Reply:
Since labels are optional and may not exists,  it's important to catch any issues and keep going.  So that needs to stay, but it might be more informative about what cause the labels to be skipped.  This is likely going to be the labels file not existing in s3, which is fine